File: DSA/SakshamDwivedi_590013147/Day16/D16P1.py
#Solution Function
def linear_search(arr, target):
    for i in range(len(arr)):
        if arr[i] == target:
            return i
    return -1

#Fixed Input
array = [10, 20, 30, 40, 50, 60]
print(linear_search(array, 30))

#Custom Input
array = list(float(x) for x in input("Enter the array : ").split())
tar = input("Enter the target value : ")
if tar == "":
    tar = None
else:
    tar = float(tar)
print(linear_search(array, tar))

File: DSA/SakshamDwivedi_590013147/Day16/D16P1.c
#include <stdio.h>

float arr[100];
float target;
int size;
int found = 0;

int main(){
    printf("Enter the size of the array : ");
    scanf("%d", &size);
    printf("Enter the elements : ");
    for(int i = 0; i < size; i++){
        scanf("%f", &arr[i]);
    }
    printf("Enter the target value : ");
    scanf("%f", &target);

    for(int i = 0; i < size; i++){
        if (arr[i] == target){
            printf("%d", i);
            found = 1;
            break;
        }
    }
    if(found == 0){
        printf("-1");
    }
}
